File: chapter7/typescript/OrderMenuRequest.ts


/**
 * client menu order requests, they can order multiple noodle, sushi, tempura items
 *
 * @version 1.0.0
 * @Author Bridget Wu
 */
 export class OrderMenuRequest {
    protected noodleItems: String[] = [];
    protected sushiItems: String[] = [];
    protected tempuraItems: String[] = [];

    /**
     * get all noodle orders
     *
     * @return
     */
    public getNoodleItems(): String[]{
        return this.noodleItems;
    }
    
    /**
     * add a noodle order
     *
     * @param item
     */
    public addNoodleItem(item: String): void {
        this.noodleItems.push(item);
    }
    /**
     * get all sushi orders
     *
     * @return
     */
     public getSushiItems(): String[]{
        return this.sushiItems;
    }
    
    /**
     * add a sushi order
     *
     * @param item
     */
    public addSushiItem(item: String): void {
        this.sushiItems.push(item);
    }
    /**
     * get all tempura orders
     *
     * @return
     */
    public getTempuraItems(): String[]{
        return this.tempuraItems;
    }
        
    /**
     * add a tempura order
     *
     * @param item
     */
    public addTempuraItem(item: String): void {
        this.tempuraItems.push(item);
    }
}

File: chapter7/java/designpattern/src/main/java/com/designpattern/OrderMenuRequest.java
package com.designpattern;

import java.util.ArrayList;
import java.util.List;

/**
 * client menu order requests, they can order multiple noodle, sushi, tempura items
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class OrderMenuRequest {

    private List<String> noodleItems = new ArrayList<>();

    private List<String> sushiItems = new ArrayList<>();

    private List<String> tempuraItems = new ArrayList<>();

    /**
     * get all noodle orders
     *
     * @return
     */
    public List<String> getNoodleItems() {
        return noodleItems;
    }

    /**
     * add a noodle order
     *
     * @param item
     */
    public void addNoodleItem(String item) {
        noodleItems.add(item);
    }

    /**
     * get all sushi orders
     *
     * @return
     */
    public List<String> getSushiItems() {
        return sushiItems;
    }

    /**
     * add a sushi order
     *
     * @param item
     */
    public void addSushiItem(String item) {
        sushiItems.add(item);
    }

    /**
     * get all tempura orders
     *
     * @return
     */
    public List<String> getTempuraItems() {
        return tempuraItems;
    }

    /**
     * add a tempura order
     *
     * @param item
     */
    public void addTempuraItem(String item) {
        tempuraItems.add(item);
    }
}
